nevergrad/functions/multiobjective.py

```diff
@@ -0,0 +1,395 @@
+#    This file is part of DEAP.
+#
+#    Copyright (C) 2010 Simon Wessing
+#    TU Dortmund University
+#
+#    In personal communication, the original authors authorized DEAP team
+#    to use this file under the Lesser General Public License.
+#
+#    You can find the original library here :
+#    http://ls11-www.cs.uni-dortmund.de/_media/rudolph/hypervolume/hv_python.zip
+#
+#    DEAP is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Lesser General Public License as
+#    published by the Free Software Foundation, either version 3 of
+#    the License, or (at your option) any later version.
+#
+#    DEAP is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Lesser General Public License for more details.
+#
+#    You should have received a copy of the GNU Lesser General Public
+#    License along with DEAP. If not, see <http://www.gnu.org/licenses/>.
+#
+#    Modified by O. Teytaud as needed for Nevergrad.
+
+from ..instrumentation import InstrumentedFunction
+from math import log, floor
+import random
+import warnings
+
+import numpy
+
+def hypervolume(pointset, ref=None):
+    """Compute the absolute hypervolume of a *pointset* according to the
+    reference point *ref*.
+    """
+    if ref is None:
+        ref = numpy.array([3000000] * len(pointset[0]))
+    warnings.warn("Falling back to the python version of hypervolume "
+        "module. Expect this to be very slow.", RuntimeWarning)
+    hv = _HyperVolume(ref)
+    return hv.compute(pointset)
+
+def multiobjective_minimization(functions, bad_values) --> InstrumentedFunction:
```

Comment:
I don't understand what are `bad_values`
When I understand all this better I can restructure and make it more compatible with the rest of the code.

Reply:
bad_values[i] is the threshold at which the objective function functions[i] becomes so bad that we don't want to keep any point, even if they are good for other objective function values.